R/explain.R

```diff
@@ -402,8 +485,37 @@ remove_outputs_to_pass_tests <- function(output) {
       NULL
   }
 
-  # Remove the `regression` parameter from the output list when we are not doing regression
-  if (isFALSE(output$internal$parameters$regression)) output$internal$parameters$regression <- NULL
+  # Removing the fit times for regression surrogate models
+  if ("regression_surrogate" %in% output$internal$parameters$approach) {
+    # Deletes the fit_times for approach = regression_surrogate to make tests pass.
+    # In the future we could delete this only when a new argument in explain called testing is TRUE
+    output$internal$objects$regression.surrogate_model$pre$mold$blueprint$recipe$fit_times <- NULL
+  }
+
+  # Delete the saving_path
+  output$internal$parameters$adaptive_arguments$saving_path <- NULL
 
   return(output)
 }
+
+get_prev_internal <- function(prev_shapr_object,exclude_parameters = c("max_n_coalitions","adaptive_arguments","seed")){
+  cl <- class(prev_shapr_object)[1]
+
+  if(cl=="character"){
+    internal <- readRDS(file = prev_shapr_object)# Already contains only "parameters" and "iter_list"
+  } else if(cl=="shapr"){
+    internal <- prev_shapr_object$internal[c("parameters","iter_list")]
+  } else{
+    stop("Invalid `shapr_object` passed to explain(). See ?explain for details.")
+  }
+
+  if(length(exclude_parameters)>0){
+    internal$parameters[exclude_parameters] <- NULL
+  }
+
+  iter <- length(internal$iter_list)
+  internal$iter_list[[iter]]$converged <- FALSE # hard setting the convergence parameter
+
+  return(internal)
+}
```

Comment:
This function could be cleaned up a little and I'm happy to do it if I get the 🆗.

Reply:
Sure, go ahead @aredelmeier :-)